node_modules/oae-content/lib/internal/util.js

```diff
@@ -104,10 +106,35 @@ var augmentContent = module.exports.augmentContent = function(ctx, content, dura
             content.previews.wideUrl = getSignedDownloadUrl(ctx, content.previews.wideUri, duration, offset);
             delete content.previews.wideUri;
         }
+
+        if (content.previews.video) {
+            content.previews.video = _getSignedVideoUrl(ctx, content.previews.key);
+        }
     }
 };
 
 /**
+ * Generate a signed url that can be used to show video previews in the browser for a video file stored in AWS S3
+ *
+ * @param  {Context}    ctx         Standard context object containing the current user and the current tenant
+ * @param  {String}     key         The name of the file stored in S3
+ * @return {String}                 The signed S3 url that can be used to view the video
+ */
+var _getSignedVideoUrl = function(ctx, key) {
+    var config = ZencoderUtil.getConfig(ctx.tenant().alias);
+
+    AWS.config.update({
+        'accessKeyId': config['amazons3-access-key'],
+        'secretAccessKey': config['amazons3-secret-key'],
+        'region': config['amazons3-region']
+    });
```

Comment:
I agree we need to move to something supported, though if it can't be configured in a multi-threaded environment (using multi-threading loosely here for multiple asynchronous calls), then it's not a real option.

If we aren't using the library elsewhere, it can still cause issues for multiple concurrent calls to get a signed URL, and it will be a really tricky issue to catch if later down the line we use it and the developer and reviewer don't catch this global config gotcha.